Jamfile

```diff
@@ -376,6 +377,12 @@ rule building ( properties * )
 		}
 	}
 
+	if [ configure.builds /boost/architecture//x86 : $(properties) : x86 ]
+		&& ! <target-os>windows in $(properties)
```

Comment:
I thought about it (passing the flag) and was expecting some comment.

I couldn't find any documentation about `gcc` using it implicitly. Actually, even for the built-in call, the instruction is not generated if this flag is not passed. What's worse, the default implementation in assembly of this built-in without the flag, is slower that the pure C++ implementation (this is all anecdotal and from internet).

In any case, I think I can left the `x86` part out of this PR and the best solution is coding it in assembly.